src/include/uapi/ipc/topology.h

```diff
@@ -113,9 +113,11 @@ struct sof_ipc_comp_config {
 	uint32_t preload_count;	/**< how many periods to preload */
 	uint32_t frame_fmt;		/**< SOF_IPC_FRAME_ */
 	uint32_t xrun_action;
+	uint16_t flavour;
```

Comment:
Could you confirm that flavor becomes a mandatory field for each end every component ? Does it mean that list of components is fixed and upcoming components will be just a different flavors of existing ones ?

Reply:
I am also less and less clear on this one. This could be a subtype UUID. It's not scalable if for every addition of a new flavor we have to change the firmware and kernel files.

Reply:
The intention here is that common kernel widgets like mux, mixer and volume can have more than one implementation in FW depending on use case. This allows topology to select the specific version of common component, as the kernel does not care and treats them as regular widgets of that type. 
@lbetlej yes it would be mandatory and defaulted to type 0
@plbossart yes it's a subtype ID, but no kernel changes needed. FW does need all types built in today but not in future with runtime FW modules